qualang_tools/control_panel/video_mode/data_acquirers.py

```diff
@@ -0,0 +1,341 @@
+
+            with stream_processing():
+                streams = {
+                    "I": IQ_streams["I"].buffer(self.x_axis.points * self.y_axis.points),
+                    "Q": IQ_streams["Q"].buffer(self.x_axis.points * self.y_axis.points),
+                }
+                combined_stream = None
+                for var in self.stream_vars:
+                    if combined_stream is None:
+                        combined_stream = streams[var]
+                    else:
+                        combined_stream = combined_stream.zip(streams[var])
+                combined_stream.save("combined")  # type: ignore
+        return prog
+
+    def process_results(self, results: Dict[str, Any]) -> np.ndarray:
+        """Process the results from the device.
+
+        This method processes the results from the device and returns a 2D array.
+        The class variable `result_type` determines the type of result to acquire.
+        The `scan_mode` determines the order in which the data is acquired and sorted
+        """
+        if self.result_type in ["I", "Q"]:
+            result = results[self.result_type]
+        elif self.result_type == "abs":
+            result = np.abs(results["I"] + 1j * results["Q"])
+        elif self.result_type == "phase":
+            result = np.angle(results["I"] + 1j * results["Q"])
+        else:
+            raise ValueError(f"Invalid result type: {self.result_type}")
+
+        x_idxs, y_idxs = self.scan_mode.get_idxs(x_points=self.x_axis.points, y_points=self.y_axis.points)
+        results_2D = np.zeros((self.y_axis.points, self.x_axis.points), dtype=float)
+        results_2D[y_idxs, x_idxs] = result
+
+        return results_2D
+
+    def acquire_data(self) -> np.ndarray:
+        """Acquire data from the device.
+
+        This method acquires data from the device and returns a 2D array.
+        """
+        if self.program is None:
+            self.run_program()
+
+        t0 = perf_counter()
+        results_tuple = self.job.result_handles.get("combined").fetch_all()  # type: ignore
+        self.results = dict(zip(self.stream_vars, results_tuple))  # type: ignore
+        result_array = self.process_results(self.results)
+        logging.info(f"Time to acquire data: {(perf_counter() - t0) * 1e3:.2f} ms")
+
+        return result_array
+
+    def run_program(self, verify: bool = True):
+        """Run the QUA program.
+
+        This method runs the QUA program and returns the results.
+
+        Args:
+            verify: Whether to verify that data can be acquired once started.
+        """
+        if self.program is None:
+            self.program = self.generate_program()
+
+        self.job = self.qm.execute(self.program)
+
+        if not verify:
+            return
+
+        # Wait until one buffer is filled{
+        self.job.result_handles.get("combined").wait_for_values(1)  # type: ignore
+
+
+class OPXQuamDataAcquirer(OPXDataAcquirer):
```

Comment:
Is this one functional?
If so, docstrings? Readme?
If not, let's remove it? Or return a nonimplemented error?